### DISCUSSION TITLE
Idea: more flexible bundler config

### DISCUSSION THREAD

**Discussion by hurricup** [ruby/rubygems#6135] [Ideas]
As far as I get it, current bundler config (above project level) is shared between all rubies I might have.
But sometimes, some options are necessary only on particular version of ruby.

E.g. I encountered an issue installing nokogiri 1.6.6.4 on macOs for ruby 2.3 and 2.4. And this may be fixed with `bundle config build.nokogiri --use-system-libraries`

I'm not sure if it will harm other versions, where nokogiri installs fine without it, but I can easily imagine case when something like this happen. 

So, it feels that some bundler options could depend on ruby version in use, and some, like `build` option, could be more precise, like include gem version constraints. 

Or, another ruby-level config could be introduced to solve part of the issue. This feels like cheap solution.

**Comment by hurricup** [comment]
Or even not ruby version, but ruby path. I may have several rubies of the same version, like system and rvm one and they have their quirks

**Reply by hurricup** [reply]
another case is puma and openssl. Older versions of puma are not smart enough to use the same openssl as used to build ruby. So on ubuntu 22.04 and ruby 3.0 you should have explict option to use the openssl1.1, otherwise some versions of puma will use the default openssl 3.0 and it will fail in the end. but I can't do this globally, because bundler config is the same for all ruby versions and ruby 3.1+ should use openssl 3+

**Reply by MSP-Greg** [reply]
> Older versions of puma

Re Puma, maybe the changes could be backported.  Hasn't been discussed.  What older versions would you like to see updated?

There have been very few updates to older Puma versions.

**Reply by hurricup** [reply]
well this request is not about the puma. it was 5.6.smth, have no before me right now. But it is just an example I encountered this time. It's not always possible to backport something, gem in question may be a dependency with some strict bound of other gem, etc. The point is that bundler setting may be different for different versions of ruby, and, honestly, for different versions of gems. And would be nice to be able to handle it.
